bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
```

Comment:
Let's separate core services from the Symfony integration and restructure them like this:

```
Core/
    VisibilityHandler/
        Content.php
        ContentAndLocation.php
        Location.php
        ObjectState.php
        Section.php
    VisibilityHandler.php
    ScheduledVisibilityService.php
    Enum/
        VisibilityUpdateResult.php
    Exception/
        InvalidStateException.php
    Configuration.php
```

Note that handler implementation doesn't do anything in regard to scheduling, it's only about visibility, so it's better named just `VisibilityHandler`. Also, convert the interface to an abstract class. Read [this post](https://qafoo.com/blog/026_abstract_classes_vs_interfaces.html) for more info on when to choose each (exceptions apply :).

Reply:
fixed in c06e4d7c45a942b756c29b0e9f94a595242396f8